packages/@o3r/components/builders/metadata-check/index.it.spec.ts

```diff
@@ -0,0 +1,407 @@
+  createConfig('@o3r/lib7', 'NewConfig7', ['prop7']),
+  createConfig('@new/lib8', 'MyConfig8', ['prop8']),
+  createConfig('@new/lib9', 'MyConfig9', ['prop9'])
+];
+
+function writeFileAsJSON(path: string, content: object): void {
+  writeFileSync(path, JSON.stringify(content), { encoding: 'utf8' });
+}
+
+function getLatestPackageVersion(packageRef: string, execOptions: ExecSyncOptions): string | undefined {
+  try {
+    const result = packageManagerInfo(packageRef, ['--json'], execOptions);
+    const latestVersion = JSON.parse(result)['dist-tags']?.latest;
+    return latestVersion && coerce(latestVersion).toString() || undefined;
+  } catch (error) {
+    return undefined;
+  }
+}
+
+const initTest = (
+  builderOptions: ConfigMigrationMetadataCheckBuilderSchema,
+  newMetadata: ComponentConfigOutput[],
+  migrationData: MigrationFile<MigrationConfigData>,
+  packageNameSuffix: string
+) => {
+
+  const { workspacePath, projectName, projectPath, o3rVersion } = o3rEnvironment.testEnvironment;
+  const execAppOptions = { ...getDefaultExecSyncOptions(), cwd: projectPath };
+  const execAppOptionsWorkspace = { ...getDefaultExecSyncOptions(), cwd: workspacePath };
+  packageManagerAdd(`@o3r/components@${o3rVersion}`, execAppOptionsWorkspace);
+  packageManagerAdd(`@o3r/extractors@${o3rVersion}`, execAppOptionsWorkspace);
+  const packageJsonPath = join(projectPath, 'package.json');
+  const angularJsonPath = join(workspacePath, 'angular.json');
+  const metadataPath = join(projectPath, metadataFileName);
+  const migrationDataPath = join(projectPath, migrationDataFileName);
+
+  // Add builder options
+  const angularJson = existsSync(angularJsonPath) ? JSON.parse(readFileSync(angularJsonPath, { encoding: 'utf8' }).toString()) : undefined;
+  const builderConfig = {
+    builder: '@o3r/components:check-config-migration-metadata',
+    options: builderOptions
+  };
+  angularJson.projects[projectName].architect['check-metadata'] = builderConfig;
+  writeFileAsJSON(angularJsonPath, angularJson);
+
+  // Add scope to project for registry management
+  let packageJson = existsSync(packageJsonPath) ? JSON.parse(readFileSync(packageJsonPath, { encoding: 'utf8' }).toString()) : undefined;
+  const packageName = `@o3r/${packageJson.name}-${packageNameSuffix}`;
+  packageJson = {
+    ...packageJson,
+    name: packageName,
+    private: false
+  };
+  writeFileAsJSON(packageJsonPath, packageJson);
+
+  // Set old metadata and publish to registry
+  writeFileAsJSON(metadataPath, previousConfigurationMetadata);
+
+  const bumpedVersion = inc(getLatestPackageVersion(packageName, execAppOptionsWorkspace) || baseVersion, 'patch');
+
+  const args = getPackageManager() === 'yarn' ? [] : ['--no-git-tag-version', '-f'];
+  packageManagerVersion(bumpedVersion, args, execAppOptions);
+  const userConfigPath = join(relative(projectPath, workspacePath), '.verdaccio/conf/.npmrc-logged');
+  packageManagerPublish(['--userconfig', userConfigPath], execAppOptions);
```

Comment:
unfortunately, the `.npmrc` in the generated project will take precedence over the `--userconfig`
we probably need to add the token to this `.npmrc`